src/third/redux-persist/createPersistor.js

```diff
@@ -49,13 +49,14 @@ export default function createPersistor (store, config) {
   async function writeOnce(state) {
     writeInProgress = true
 
-    const updatedSubstates = []
-
-    Object.keys(state).forEach((key) => {
-      if (!passWhitelistBlacklist(key)) return
-      if (lastWrittenState[key] === state[key]) return
-      updatedSubstates.push([key, state[key]])
-    });
+    // Atomically collect the subtrees that need to be written out.
+    const updatedSubstates = [];
+    for (const key of Object.keys(state)) {
+      if (state[key] === lastWrittenState[key] || !passWhitelistBlacklist(key)) {
+        continue;
```

Comment:
I think these are actually cleanest as two separate conditionals. The reasoning behind them -- what you'd say in a comment to explain them, if commenting more verbosely like that -- is pretty different.